### DISCUSSION TITLE
setBackgroundMessageHandler not called when app is background IOS

### DISCUSSION THREAD

**Discussion by ReactNativeDev2296** [invertase/react-native-firebase#6233] [Q&A]
Hello , i am using React native firebase v10 and not getting call back for setBackgroundMessageHandler

**Comment by mikehardy** [comment]
Hi there! In short, do not use v10. No effort to support older versions is made, troublshoot on current versions of software only, as we would likely just be chasing ghosts if so.

That said, this area of "trouble" (ios not receiving background messages) is typically a result of incorrect expectations around data-only messages and their delivery guarantees.

To be quite clear: data-only messages are **not reliable by design** - Apple will only make a weak effort to deliver them, based on lots of factors. Do not rely on them. If that's your problem - a data-only message not being delivered - then it's expected behavior

If you are on current versions and using notification blocks then you might try to access help again but please keep in mind that *you* must do a fair bit of preparation work to receive useful help

https://stackoverflow.com/help/how-to-ask
https://stackoverflow.com/help/minimal-reproducible-example

**Comment by ReactNativeDev2296** [comment]
Hello @mikehardy,
React native firebase messaging version 14.9.1
- below the index.js file in I added the setbackgroundhandler.  

`import messaging from '@react-native-firebase/messaging';
import { AppRegistry } from 'react-native';
import App from './App';
import { name as appName } from './app.json';
messaging().setBackgroundMessageHandler(async (remoteMessage) => {
	console.log(remoteMessage);
})

AppRegistry.registerComponent(appName, () => App);`

- I also enabled **background fetch, remote notifications, background processing** but get the remoteMessage in setbackgroundhandler and FCM also provided below body.
-`"content_available": true,
    "priority": "high"`

so please help me above issues

**Reply by mikehardy** [reply]
Without seeing the raw JSON that you send to the FCM REST API for testing, I'm not sure really (there is no substitute for that, don't try to work around it)
You should also turn on debug logging for firebase and watch the log on a device that is plugged in to your computer running the app
I probably will not be able to help you really, the typical end result of this is the issue logger discovers they are sending data only notifications on iOS and they run in to the reality that's not reliable, then they change their requirements to accept that reality
